xbmc/windowing/wayland/WinSystemWayland.cpp

```diff
@@ -313,8 +317,13 @@ bool CWinSystemWayland::CreateNewWindow(const std::string& name,
   if (!m_shellSurface)
   {
     CLog::LogF(LOGWARNING, "Compositor does not support xdg_shell protocol (stable or unstable v6) - falling back to wl_shell, not all features might work");
-    m_shellSurface.reset(new CShellSurfaceWlShell(*this, *m_connection, m_surface, name,
+    #ifdef TARGET_WEBOS
+      m_shellSurface.reset(new CShellSurfaceWebOSShell(*this, *m_connection, m_surface, name,
```

Comment:
Any reason this isn't tried first?

Reply:
@lrusak CShellSurfaceWlShell wont actually fail, but you will just end up with a blank screen.

Reply:
I mean why try after `CShellSurfaceXdgShell` ??